### ISSUE TITLE
Connection error: MongoDB can only handle 8-byte ints

### ISSUE THREAD

**Issue by 0x80** [bobthecow/genghis#92]
What does this mean? I am trying to use Genghis for the first time. I have recent versions of both the app, ruby and mongodb....

mongod --version
db version v2.4.0

ruby --version
ruby 2.0.0p0 (2013-02-24 revision 39474) [x86_64-darwin12.3.0]

genghis v2.2.2

**Comment by bobthecow** [comment]
That looks like a server issue rather than a client issue. Did you install 32-bit MongoDB instead of 64?

**Comment by bobthecow** [comment]
The only place I can find this message is here:

https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/bson/bson_ruby.rb#L447

What are you doing to get this message?

**Comment by 0x80** [comment]
Hmm, I thought it might be a conflict between an older 10gen install and my latest install using homebrew. But I don't get it. I installed with homebrew again, making sure it's 64 bit. There is something about i386 in the logs related to libboost? Below is my log.

I get this error when I launch the Genghis app and it tries to connect to the only server I have set up, on my localhost.

Mon Mar 25 12:04:59.597 [initandlisten] MongoDB starting : pid=29003 port=27017 dbpath=/usr/local/var/mongodb 64-bit host=Me-Macbook-Pro.local
           12:04:59.798 [initandlisten] 
           12:04:59.799 [initandlisten] *\* WARNING: soft rlimits too low. Number of files is 256, should be at least 1000
           12:04:59.799 [initandlisten] db version v2.4.1
           12:04:59.799 [initandlisten] git version: 1560959e9ce11a693be8b4d0d160d633eee75110
           12:04:59.799 [initandlisten] build info: Darwin bs-osx-106-x86-64-1.local 10.8.0 Darwin Kernel Version 10.8.0: Tue Jun  7 16:33:36 PDT 2011; root:xnu-1504.15.3~1/RELEASE_I386 i386 BOOST_LIB_VERSION=1_49
           12:04:59.799 [initandlisten] allocator: system
           12:04:59.799 [initandlisten] options: { bind_ip: "127.0.0.1", config: "/usr/local/etc/mongod.conf", dbpath: "/usr/local/var/mongodb", logappend: "true", logpath: "/usr/local/var/log/mongodb/mongo.log" }
           12:04:59.800 [initandlisten] journal dir=/usr/local/var/mongodb/journal
           12:04:59.800 [initandlisten] recover : no journal files present, no recovery needed
           12:04:59.923 [initandlisten] waiting for connections on port 27017
           12:04:59.923 [websvr] admin web console waiting for connections on port 28017

**Comment by 0x80** [comment]
I updated libboost on my system and made sure it's universal binary. Still mongodb installs with this i386 release via homebrew. I have no version 1.49 on my system. So maybe it has nothing to do with it, but I now ran out of ideas. Any suggestions?

**Comment by 0x80** [comment]
I have figured it out. It was a problem with the ruby mongo driver.

Apparently I had to install the bson_ext gem in order for Genghis to work. 

This lead me to a gem install error:
unable to convert "\xCF" from ASCII-8BIT to UTF-8 for ext/bson_ext/bson_ext/cbson.bundle

I found that this is an issue related to the OSX terminal
http://stackoverflow.com/questions/4975973/gem-update-unable-to-convert-xe7-to-utf-8-in-conversion-from-ascii-8bit-to-u

Adding this export to my ~/.profile
export LC_CTYPE="utf-8"

And restarting Terminal (sourcing doesn't work when using sudo?)

I could then succesfully install the gem:
sudo gem install bson_ext

Restarted Genghisapp and now I'm connected!
